ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/Transaction.java

```diff
@@ -495,8 +498,8 @@ public Transaction signAndBuild(final SECP256K1.KeyPair keys) {
     }
 
     protected SECP256K1.Signature computeSignature(final SECP256K1.KeyPair keys) {
-      final OptionalInt optionalChainId =
-          chainId > 0 ? OptionalInt.of(chainId) : OptionalInt.empty();
+      final Optional<BigInteger> optionalChainId =
+          chainId.signum() == 1 ? Optional.of(chainId) : Optional.empty();
```

Comment:
With the changes above you then don't need this and can just use `chainId` directly since it's already an optional.

Reply:
done